addons/netfox/serialization/properties-serializer.gd

```diff
@@ -0,0 +1,41 @@
+extends Object
+class_name PropertiesSerializer
+
+static func deserialize_multiple_properties(serialized_input: PackedByteArray, auth_input_pros: Array[PropertyEntry]) -> Dictionary:
+	var reconstructed_dictionary: Dictionary = {} #Exclusively for that tick
+	
+	print("Deserializing multiple properties of serialized input %s  and array %s" % [serialized_input, auth_input_pros])
+	var property_type: Variant.Type
+	var property_type_byte_size: int
+	var serialized_property: PackedByteArray
+	var input_byte_index: int = 0
+	for picked_property_entry in auth_input_pros:
+		property_type = picked_property_entry.type
+		property_type_byte_size = ValueToBytes.get_byte_size(property_type)
+		serialized_property = serialized_input.slice(input_byte_index, input_byte_index + property_type_byte_size)
+		
+		var picked_value = ValueToBytes.deserialize(serialized_property, property_type)
+		input_byte_index += property_type_byte_size
+		
+		print("path is %s and value is %s" % [picked_property_entry._path, picked_value])
+		reconstructed_dictionary[picked_property_entry._path] = picked_value
+	return reconstructed_dictionary
+
+## Extract but in serialized format!
```

Comment:
The comment doesnt explain itself (should I delete it?) so i want to explain it fully:

See PropertySnapshot.extract() function. This function is basically the same but it returns the Dictionary in PackedByteArray format.